### PR TITLE
tests: fix for preseeding failures

### PR DESCRIPTION
**Pull request by zyga**
This patch fixes the various preseed tests in two ways:

- re-packing the core snap is no longer done (preseed), both 19.10 and 20.40 use core18 and snapd for seeds.
- restore sections were not sourcing the `preseed.sh` script
- prepare now mounts, restore now unmounts, for regularity
- mounted qemu-nbd no longer hangs ssh on failure and is run as a systemd service
- removed preseed-snapd-snap test as the functionality is now covered by the main preseed test (no need to inject/hack core/snapd snaps for preseeding).

### CODE REVIEW

**Comment by zyga** [comment]
@stolowski this hangs in the preseed tests, oh well :-(

File: tests/lib/preseed.sh [hunk 1]
```diff
@@ -43,17 +43,33 @@ umount_ubuntu_image() {
...
+        cp /usr/lib/snapd/snapd squashfs-root/usr/lib/snapd/snapd
+        # XXX to satisfy version check; this will go away once preseeding
+        # is available in 2.44
+        echo "VERSION=2.44.0" > squashfs-root/usr/lib/snapd/info
+        rm "$SNAPD_IMAGE"
+        #shellcheck source=tests/lib/snaps.sh
+        . "$TESTSLIB"/snaps.sh
+        mksnap_fast squashfs-root "$SNAPD_IMAGE"
+        sed -i "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/seed.yaml" -E -e 's/^(\s+)name: snapd/\1name: snapd\n\1unasserted: true/'
+    fi
```
Comment (jhenstridge) [inline tests/lib/preseed.sh:74]: These two blocks look pretty much identical.  Could you collapse them back down with a loop like `for snap in core snapd; do ...`?

Comment (zyga) [reply tests/lib/preseed.sh:77]: Yeah, sure thing.

Comment (zyga) [reply tests/lib/preseed.sh]: Done

File: tests/main/preseed/task.yaml [hunk 2]
```diff
@@ -42,51 +43,104 @@ execute: |
...
+    MATCH "Do .+ run-hook +Run configure hook of \"lxd\" snap if present" < tasks.log
+    MATCH "Do .+ run-hook +Run health check of \"lxd\" snap" < tasks.log
+    MATCH "Do .+ mark-seeded +Mark system seeded" < tasks.log
+    MATCH "Do .+ run-hook +Run install hook of \"snapd\" snap if present" < tasks.log
+    MATCH "Do .+ start-snap-services +Start snap \"snapd\" \([0-9]+\) services" < tasks.log
+    MATCH "Do .+ run-hook +Run configure hook of \"core\" snap if present" < tasks.log
+    MATCH "Do .+ start-snap-services  +Start snap \"core18\" \([0-9]+\) services" < tasks.log
+    MATCH "Do .+ run-hook +Run health check of \"core18\" snap" < tasks.log
+    MATCH "Do .+ run-hook +Run install hook of \"lxd\" snap if present" < tasks.log
+  else
```
Comment (mvo5) [inline tests/main/preseed/task.yaml:99]: nitpick: I wonder if we should be explicit here that this applies to 19.10 and have an else that just fails if we run it on something unexpected?

Comment (stolowski) [reply tests/main/preseed/task.yaml:99]: Good idea, will do.

File: tests/main/preseed-reset/task.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ description: |
   can be undone with --reset flag.
 
 # TODO: reenable for ubuntu-20.04-*, that now uses the snapd snap
```
Comment (anonymouse64) [inline tests/main/preseed-reset/task.yaml:6]: this TODO can be deleted, no?

File: tests/lib/preseed.sh [hunk 4]
```diff
@@ -13,9 +13,12 @@ mount_ubuntu_image() {
         modprobe nbd
     fi
 
-    qemu-nbd -c /dev/nbd0 "$CLOUD_IMAGE"
+    # Run qemu-ndb as a service, so that it does not interact with ssh
+    # stdin/stdout it would otherwise inherit from the spread session.
+    systemd-run --system --service-type=forking --unit=qemu-ndb-preseed.service "$(command -v qemu-nbd)" --fork -c /dev/nbd0 "$CLOUD_IMAGE"
```
Comment (anonymouse64) [inline tests/lib/preseed.sh:18]: nice fix!

File: tests/main/preseed-reset/task.yaml (hunk 3 above)
Comment (zyga) [inline tests/main/preseed-reset/task.yaml:6]: Let's remove this in the next pass.

**Pull request merged**